File: go/permutations.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func printPermutations(a []string, n int) {
	if n <= 0 {
		fmt.Println(strings.Join(a, "\t"))
	} else {
		for i := 0; i < n; i++ {
			printPermutations(a, n-1)
			if n % 2 == 0 {
				a[0], a[n] = a[n], a[0]
			} else {
				a[i], a[n] = a[n], a[i]
			}
		}
		printPermutations(a, n-1)
	}
}

func main() {
	args := os.Args[1:]
	printPermutations(args, len(args)-1)
}

File: csharp/Permutations.cs
using System;

static class Permutations
{
    static void PrintPermutations(string[] a, int n)
    {
        if (n <= 0)
        {
            Console.WriteLine(string.Join("\t", a));
        }
        else
        {
            for (int i = 0; i < n; i++)
            {
                PrintPermutations(a, n - 1);
                int j = n % 2 == 0 ? 0 : i;
                (a[j], a[n]) = (a[n], a[j]);
            }
            PrintPermutations(a, n - 1);
        }
    }

    static void Main(string[] args)
    {
        if (args.Length < 1)
        {
            throw new Exception("ERROR: There must be at least 1 argument.");
        }
        PrintPermutations(args, args.Length - 1);
    }
}